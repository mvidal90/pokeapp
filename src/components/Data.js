import React from 'react';

export const Data = ({ data }) => {
    return (
        <p>
            { data }
        </p>
    )
}
