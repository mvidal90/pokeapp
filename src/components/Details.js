import React from 'react';
import { Data } from './Data';

export const Details = ({ abilities, description, lg, types }) => {
    return (
        <div>
            <h5>
                {
                    lg === 'es' ? 'Tipo' : 'Type' 
                }
            </h5>
            {
                types.map( type => (
                    <Data
                        data = { type.name } 
                        key = { type.name }
                        {...type}
                    />
                )) 
            }
            <h5>
                {
                    lg === 'es' ? 'Habilidades' : 'Abilities' 
                }
            </h5>
            {
                abilities.map( ability => (
                    <Data
                        data = { ability.name }
                        key = { ability.name }
                        {...ability}
                    />
                )) 
            } 
            <h5>
                {
                    lg === 'es' ? 'Descripci√≥n' : 'Description' 
                }
            </h5>
            <p>
                {
                    description.replace('',' ')
                }
            </p>
        </div>
    )
}
