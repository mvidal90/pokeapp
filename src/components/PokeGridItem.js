import React, { useState } from 'react'
import { Details } from './Details';
import { useFetchInfoPoke } from '../hooks/useFetchInfoPoke';
import { useFetchDescription } from '../hooks/useFetchDescription';

import { Col, Modal } from 'react-bootstrap'

export const PokeGridItem = ({ lg, name, url}) => {

    const [show, setShow] = useState( false );
    const { id, abilities, types } = useFetchInfoPoke( url );
    const { description } = useFetchDescription( id, lg );

    
    const urlImg = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${ id }.svg`;

    return (
        <>
            {
                id !== undefined &&
                <Col 
                    className = "card"
                    onClick = { () => setShow(true) } 
                    sm = { 4 }
                    md = { 3 }
                    lg = { 2 }
                >
                    <div className="image">
                        <img src={ urlImg } alt={ name } />
                    </div>
                    <h3>
                        { name }
                    </h3>
                </Col>
            }
            <Modal
                show = { show }
                onHide = { () => setShow(false) }
            >
                <div className = "justify-content-md-center align-self-center landscape">
                    <img src={ urlImg } alt={ name } className = "imgModal" />
                </div>
                <Modal.Body>
                    <h4>{ name }</h4>
                    <Details 
                        abilities= { abilities } 
                        description = { description } 
                        lg = { lg }
                        types = { types }
                    />
                </Modal.Body>
            </Modal>
        </>
    )
}
