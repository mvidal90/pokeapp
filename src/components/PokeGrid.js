import React, { useState } from 'react'
import { PokeGridItem } from './PokeGridItem';
import { useFetchPoke } from '../hooks/useFetchPoke';

import { Col, Row } from 'react-bootstrap';

export const PokeGrid = ( { lg, initialOffset = 0 } ) => {

    const [ page, setPage] = useState( 1 );
    const [ offset, setOffset] = useState( initialOffset );
    const { poke } = useFetchPoke( offset );

    const handleChange = ( e ) => {
        setPage(e.target.value);
    }

    const increment = ( e ) => {
        e.preventDefault();

        setPage( parseInt(page) + 1 );
        setOffset( offset + 5 );
    }
    
    const decrement = ( e ) => {
        e.preventDefault();

        setPage( page - 1 );
        setOffset( offset - 5 )
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (page > 0 && page < 131) {
            setPage( page );
            setOffset( (page - 1) * 5 );            
        }
    }

    return (
        <>
            <Row className="justify-content-md-center align-self-center">
                {
                    poke.map( pokemon => (
                        <PokeGridItem
                            lg = { lg } 
                            key = { pokemon.name }
                            {...pokemon}
                        />
                    )) 
                }
            </Row>
            <Row className="justify-content-md-center align-self-end">
                <Col 
                    xs = { 6 } 
                    sm = { 4 }
                    lg = { 2 }
                >
                    { page > 1 &&  
                        <button
                            className = "pg"
                            onChange = { handleChange }
                            onClick = { decrement }
                        >
                            {
                                lg === 'es' ? 'Anterior' : 'Previus'
                            }
                        </button>
                    }
                </Col>
                <Col 
                    xs = { 6 }
                    sm = { 4 }
                    lg = { 2 }
                >
                    { page < 131 &&
                        <button
                            className = "pg"
                            onChange= { handleChange }
                            onClick= { increment }
                        >
                            { lg === 'es' ? 'Siguiente' : 'Next' }
                        </button>
                    }
                </Col>
            </Row>
            <Row>
                <Col className = "pagination ml-auto" xs = {2} sm = {2} md = { 2 } lg = { 1 }>
                    <p className = "ml-auto"> { lg === 'es' ? 'PÃ¡gina' : 'Page' } </p>
                </Col>
                <Col xs = {2} sm = {2} md = { 2 } lg = { 1 }>
                    <form onSubmit={ handleSubmit }>
                        <input 
                            name = "Page"
                            onChange = { handleChange }
                            type = "number"
                            value = { page }
                        />
                    </form>
                </Col>
                <Col className= "pagination" xs = {2} sm = {2} md = { 2 } lg = { 1 }>
                    <p> { lg === 'es' ? 'de 130' : 'of 130' } </p>
                </Col>
            </Row>
        </>
    );
}
