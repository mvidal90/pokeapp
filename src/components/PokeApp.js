import React, { useState } from 'react';
import { PokeGrid } from './PokeGrid';

import { Container, Row } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';

import arg from '../img/argentina.svg';
import usa from '../img/united-states.svg'

export const PokeApp = () => {

    const [language, setLanguage] = useState('es');

    return (
        <>
            <div className = "language">
                <h2>PokeApp</h2>
                <div
                    className = "lgGrid ml-auto"
                    onClick = { () => setLanguage('es') }
                >
                    <img 
                        className = "lg"
                        src = { arg }
                        alt = "Cambiar idioma al español"
                    />
                </div>
                <div
                    className = "lgGrid"
                    onClick = { () => setLanguage('en') }
                >
                    <img 
                        className = "lg"
                        src = { usa }
                        alt = "Change language to English"
                    />
                </div>
            </div>
            <Container>
                <Row className = "justify-content-md-center align-self-center">
                    <h3>{ language === 'es' ? 'Encontrá tu POKÉMON!!' : 'Find your POKÉMON!!'   }</h3>
                </Row>
                <PokeGrid lg = { language }/> 
            </Container>
        </>
    )
}
