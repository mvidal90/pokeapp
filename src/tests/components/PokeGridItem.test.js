import React from 'react';
import { shallow } from 'enzyme';
import { PokeGridItem } from '../../components/PokeGridItem';

describe('Pruebas con <PokeGridItem />', () => {

    const language = 'es';

    const name = 'Un pokemon';
    
    const url = 'https://localhost/algo.jpg';

    const wrapper = shallow( <PokeGridItem lg = { language } name={ name } url={ url }/>)

    test('Debe mostrarse correctamente', () => {

        expect( wrapper ).toMatchSnapshot(); 
        expect( wrapper.find('Modal').length).toBe(1);
        expect( wrapper.find('Details').length).toBe(1);
        expect( wrapper.find('h4').length).toBe(1);


    })

})
