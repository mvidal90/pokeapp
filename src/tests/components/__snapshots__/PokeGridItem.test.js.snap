// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <PokeGridItem /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    dialogAs={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    enforceFocus={true}
    keyboard={true}
    onHide={[Function]}
    restoreFocus={true}
    show={false}
  >
    <div
      className="justify-content-md-center align-self-center landscape"
    >
      <img
        alt="Un pokemon"
        className="imgModal"
        src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/undefined.svg"
      />
    </div>
    <ModalBody>
      <h4>
        Un pokemon
      </h4>
      <Details
        lg="es"
      />
    </ModalBody>
  </Modal>
</Fragment>
`;
